packages/salesforcedx-utils-vscode/src/services/telemetry.ts

```diff
@@ -138,63 +137,26 @@ export class TelemetryService {
     const isDevMode =
       extensionContext.extensionMode !== ExtensionMode.Production;
 
-    // TelemetryReporter is not initialized if user has disabled telemetry setting.
-    if (this.reporters.length === 0 && (await this.isTelemetryEnabled())) {
-      if (!isDevMode) {
-        console.log('adding AppInsights reporter.');
-
-        this.reporters.push(
-          new AppInsights(
-            this.getTelemetryReporterName(),
-            this.version,
-            this.aiKey,
-            true
-          )
-        );
-
-        // Assuming this fs streaming is more efficient than the appendFile technique that
-        // the new TelemetryFile reporter uses, I am keeping the logic in place for which
-        // reporter is used when.  The original log stream functionality only worked under
-        // the same conditions as the AppInsights capabilities, but with additional configuration.
-        if (LogStreamConfig.isEnabledFor(this.extensionName)) {
-          this.reporters.push(
-            new LogStream(
-              this.getTelemetryReporterName(),
-              LogStreamConfig.logFilePath()
-            )
-          );
-        }
+    if(this.reporters.length === 0) {
+      if (this.isInternal) {
```

Comment:
@nkelleysf I understand the need to override `this.isTelemetryEnabled()` so that all sf employees are "accepting" telemetry reporting. 

Seems to me that the right place to add in internal check would be in `isTelemetryEnabled` and not here. I believe we will get the same results.

Reply:
I do not disagree, however digging into the codebase and being newer to the repo I found it difficult to follow if telemetry is truly enabled or not. In this case it made sense to save the time and bug risk to override if the user is internal.

Considering the complexity and potential combinations, @gbockus-sf and myself briefly discussed adding a method that would provide a clear understanding of the state of some of these checks such as isDev, isInternal, isTelemetryEnabled.